instrumentation/kafka-streams-0.11/src/main/java/io/opentelemetry/auto/instrumentation/kafka_streams/KafkaStreamsDecorator.java

```diff
@@ -44,10 +43,20 @@ protected String getSpanType() {
     return SpanTypes.MESSAGE_CONSUMER;
   }
 
+  public String spanNameForConsume(final StampedRecord record) {
+    if (record == null) {
+      return null;
+    }
+    final String topic = record.topic();
```

Comment:
just curious (not trying to be critical), but why wouldn't you put the `else` block in this case, but you would below?

Reply:
My logic :smile: is that at the top of a method, you are short-circuiting execution.

At the bottom of the method you are returning a value.

Reply:
ok! I vaguely see the distinction, but don't like sliding down that slope.  ;)